railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -88,6 +96,29 @@ def parse_type_and_options(type)
               options = Hash[provided_options.map { |opt| [opt.to_sym, true] }]
 
               return type, options
+            when /(.+)\{(.+)\}/
+              type = $1
+              # regex to match commas that are not inside quotes.
+              # this allows values like for `default` that are strings with commas
+              # e.g. `text{default='hello, world!'}`
+              options = $2.split(/,(?![^'"]*['"][^'"]*$)/)
+                          .filter_map do |opt|
+                key, val = opt.split("=")
+                next unless AVAILABLE_OPTIONS.include?(key)
+                val = case val
+                      when "true" then true
+                      when "false" then false
+                      when "[]" then []
+                      when "{}" then {}
+                      when /^['"].*['"]$/ then val[1...-1]
+                      when nil then true
+                      when /\A\d+\z/ then Integer(val)
+                      else val
+                end
+                [key.to_sym, val]
```

Comment:
Can we just use `key.to_sym, val` here (without []) ?

Reply:
I don't believe so. I believe you only have that option when using a keyword like `return`

Reply:
@fractaledmind You're correct. So in this case, should we use `return key.to_sym, val` as i guess it's standard Ruby for multiple return value ?